PrivateCloud.DiagnosticInfo/PrivateCloud.DiagnosticInfo.psm1

```diff
@@ -5086,16 +5084,72 @@ WARNING: the SMB Client is receiving RDMA disconnects. This is an error whose ro
 
     $w = @"
 WARNING: the SMB Client is receiving RDMA connect errors. This is an error whose root
 `t cause may be actual lack of connectivity or fundamental problems with the RDMA
 `t network fabric. Please inspect especially if in the Last5 bucket.
 "@
 
-    $j += Start-Job -name 'SMB Connectivity Error Check - Connect Failures (Event 30803)' -InitializationScript $CommonFunc -ScriptBlock $ReportTableBlock -ArgumentList $eventlogs,30803,$CaptureDate,([ConsoleColor]'Yellow'),$w
+    $j += Start-Job -name 'SMB Connectivity Error Check - RDMA Connect Failures (Event 30803)' -InitializationScript $CommonFunc -ScriptBlock $ReportTableBlock -ArgumentList $eventlogs,30803,$CaptureDate,([ConsoleColor]'Yellow'),$w,@{'ConnectionType'='=2'}
+
+    $w = @"
+
+WARNING: the SMB Client is receiving TCP/IP disconnects. Note that
+
+`t cluster node reboots are a natural & expected source of disconnects.
+
+"@
+
+    $j += Start-Job -name 'SMB Connectivity Error Check -  TCP/IP Connect Failures (Event 30804)' -InitializationScript $CommonFunc -ScriptBlock $ReportTableBlock -ArgumentList $eventlogs,30804,$CaptureDate,([ConsoleColor]'Yellow'),$w, @{'ConnectionType'='=1'}
+
+    $w = @"
+
+WARNING: the SMB Client is receiving TCP/IP connect errors. 
+
+Please inspect especially if in the Last5 bucket.
+
+"@
+
+    $j += Start-Job -name 'SMB Connectivity Error Check - TCP/IP Connect Failures (Event 30803)' -InitializationScript $CommonFunc -ScriptBlock $ReportTableBlock -ArgumentList $eventlogs,30803,$CaptureDate,([ConsoleColor]'Yellow'),$w, @{'ConnectionType'='=1'}
+
+    $w = @"
+
+WARNING: the SMB Client is  experiencing long time to complete IO.
+
+SMB will close diconnected channels
+
+"@
+
+    $j += Start-Job -name 'SMB Connectivity Error Check - SMB IO Timeouts (Event 30809)' -InitializationScript $CommonFunc -ScriptBlock $ReportTableBlock -ArgumentList $eventlogs,30809,$CaptureDate,([ConsoleColor]'Yellow'),$w
+
+    $w = @"
+
+WARNING: the SMB Client has disconnect channels because IO took long time.
+
+`t Something is stuck somewhere in the stack, and we need a LiveDump to 
+
+`t figure it out. Look for CsvIoTM livedumps with close timestamp in the 
+
+`t folder that contains postmortem information
+
+"@
+
+    $j += Start-Job -name 'SMB Connectivity Error Check - SMB Connection Closed (Event 30823)' -InitializationScript $CommonFunc -ScriptBlock $ReportTableBlock -ArgumentList $eventlogs,30823,$CaptureDate,([ConsoleColor]'Yellow'),$w
 
+
+    $eventlogs = (dir $Path\Node_*\System.EVTX).FullName
+
+    $w = @"
+
+WARNING: Dirty Shutdowns has beend detected. Please verify, if those were intended.
+`t There could be connectivity issues due to it.
+
+"@
+
+    $j += Start-Job -name 'Dirty Shutdown Error Check - (Event 41)' -InitializationScript $CommonFunc -ScriptBlock $ReportTableBlock -ArgumentList $eventlogs,41,$CaptureDate,([ConsoleColor]'Yellow'),$w

```

Comment:
Suggest adding a reference to the system log here (e.g., 'System Log Event 41'). The SMB Connectivity events are named for the channel; granted, they could be more specifically the SMB Client Connectivity, but the text helps.